src/parlant/core/engines/alpha/engine.py

```diff
@@ -623,6 +765,36 @@ async def _load_relevant_terms(
             )
         return []
 
+    async def _utter_requests_to_guideline_propositions(
+        self,
+        actions: Sequence[UtteranceRequest],
+    ) -> Sequence[GuidelineProposition]:
+        def utterance_to_proposition(utterance: UtteranceRequest) -> GuidelineProposition:
+            ratioanles = {
+                UtteranceReason.BUY_TIME: "An external module has determined that this response is necessary, and you must adhere to it.",
+                UtteranceReason.FOLLOW_UP: "An external module has determined that this response is necessary, and you must adhere to it.",
+            }
+
+            conditions = {
+                UtteranceReason.BUY_TIME: "always",
+                UtteranceReason.FOLLOW_UP: "always",
+            }
+
+            return GuidelineProposition(
+                guideline=Guideline(
+                    id=GuidelineId(""),
```

Comment:
Maybe include at least some indicator here so that it doesn't look like a bug: `GuidelineId("<utterance-request-{i}>")` where `i` is received by enumerating `actions` (which BTW should be named `requests`)